services/graph/pkg/service/v0/sharedbyme.go

```diff
@@ -0,0 +1,232 @@
+			if err != nil {
+				g.logger.Error().Err(err).Msg("could not select next gateway client")
+				return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+			}
+			req := user.GetUserRequest{
+				UserId: s.Grantee.GetUserId(),
+			}
+			res, err := gatewayClient.GetUser(ctx, &req)
+			if err != nil {
+				return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+			}
+			user := libregraph.NewIdentityWithDefaults()
+			user.SetId(s.Grantee.GetUserId().GetOpaqueId())
+			switch res.Status.Code {
+			case rpc.Code_CODE_OK:
+				cs3User := res.GetUser()
+				user.SetDisplayName(cs3User.GetDisplayName())
+			case rpc.Code_CODE_NOT_FOUND:
+				g.logger.Warn().Str("userid", s.Grantee.GetUserId().GetOpaqueId()).Msg("User not found by id")
+			default:
+				return driveItems, errorcode.New(errorcode.GeneralException, res.Status.Message)
+			}
+
+			grantedTo.SetUser(*user)
+		case storageprovider.GranteeType_GRANTEE_TYPE_GROUP:
+			gatewayClient, err := g.gatewaySelector.Next()
+			if err != nil {
+				g.logger.Error().Err(err).Msg("could not select next gateway client")
+				return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+			}
+			req := group.GetGroupRequest{
+				GroupId: s.Grantee.GetGroupId(),
+			}
+			res, err := gatewayClient.GetGroup(ctx, &req)
+			if err != nil {
+				return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+			}
+			grp := libregraph.NewIdentityWithDefaults()
+			grp.SetId(s.Grantee.GetGroupId().GetOpaqueId())
+			switch res.Status.Code {
+			case rpc.Code_CODE_OK:
+				cs3Group := res.GetGroup()
+				grp.SetDisplayName(cs3Group.GetDisplayName())
+			case rpc.Code_CODE_NOT_FOUND:
+				g.logger.Warn().Str("groupid", s.Grantee.GetGroupId().GetOpaqueId()).Msg("Group not found by id")
+			default:
+				return driveItems, errorcode.New(errorcode.GeneralException, res.Status.Message)
+			}
+
+			grantedTo.SetGroup(*grp)
+		}
+
+		// set expiration date
+		if s.GetExpiration() != nil {
+			perm.SetExpirationDateTime(cs3TimestampToTime(s.GetExpiration()))
+		}
+
+		perm.SetGrantedToV2(grantedTo)
+		item.Permissions = append(item.Permissions, perm)
+		driveItems[resIDStr] = item
+	}
+	return driveItems, nil
+}
+
+func (g Graph) cs3PublicSharesToDriveItems(ctx context.Context, shares []*link.PublicShare, driveItems driveItemsByResourceID) (driveItemsByResourceID, error) {
+	for _, s := range shares {
+		g.logger.Debug().Interface("CS3 PublicShare", s).Msg("Got Share")
+		resIDStr := storagespace.FormatResourceID(*s.ResourceId)
+		item, ok := driveItems[resIDStr]
+		if !ok {
+			itemptr, err := g.getDriveItem(ctx, storageprovider.Reference{ResourceId: s.ResourceId})
+			if err != nil {
+				g.logger.Debug().Err(err).Interface("Share", s.ResourceId).Msg("could not stat share, skipping")
+				continue
+			}
+			item = *itemptr
+		}
+		perm := libregraph.Permission{}
+		perm.SetId(s.Id.OpaqueId)
+		perm.SetRoles([]string{"TODO:insert-role-here"})
```

Comment:
Same question as above: Is this intended? If yes: why?

Reply:
Same answer :smile: